### PR TITLE
Add fragment file region scoring matrix creation

### PR DESCRIPTION
**Pull request by nleroy917**

### CODE REVIEW

File: gtars/tests/test.rs [hunk 1]
```diff
@@ -91,6 +91,44 @@ mod tests {
...
+        let mut va = 0;
+
+        let result = parse_bed(&bed_file_string, &mut start, &mut end, &mut va).unwrap(); // this will return
+
+        let unwrapped_result = result.as_str();
+
+        assert_eq!(unwrapped_result, "chr1");
+
+        // Ensure start and end is modified via parse_bed
+        assert_eq!(start, 32481);
```
Comment (donaldcampbelljr) [inline gtars/tests/test.rs:128]: Why are these igd tests showing up here as new?

File: gtars/src/scoring/cli.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+pub fn make_fscoring_cli() -> Command {
+    Command::new(consts::FSCORING_CMD)
+        .author("Nathan LeRoy")
+        .about("Create a scoring matrix for a set of fragment files over a consensus peak set.")
+        .arg(Arg::new("fragments"))
+        .arg(Arg::new("consensus"))
+        .arg(arg!(--mode <mode>))
+        .arg(arg!(--output <output>))
+        .arg(arg!(--whitelist <whitelist>))
```
[the remark this answers is not in the document]
Comment (nleroy917) [reply gtars/src/scoring/cli.rs:19]: yeah let me remove this

File: gtars/src/scoring/cli.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            .get_one::<String>("consensus")
+            .expect("A path to a mapping file is required.");
+
+        let default_out = consts::DEFAULT_OUT.to_string();
+        let output = matches.get_one::<String>("output").unwrap_or(&default_out);
+        let mode = match matches.get_one::<String>("mode") {
+            Some(mode) => ScoringMode::from_str(mode),
+            None => Ok(DEFAULT_SCORING_MODE),
+        };
+        let mode = mode.unwrap_or(DEFAULT_SCORING_MODE);
```
[the remark this answers is not in the document]
Comment (nleroy917) [reply gtars/src/scoring/cli.rs:48]: Ah. The reason is that `ScoringMode::from_str` can return `Err`.. so I need to do a second `unwrap`... I suppose a better way might be to expand out the match arm for matching `Some(mode)`... and then trying to unwrap the results of `from_str` and warning if an unknown mode was supplied.

[the remark this answers is not in the document]
Comment (nleroy917) [reply gtars/src/scoring/cli.rs:48]: Yeah this is bad code haha. I am not only masking a user error, but also its hard to read

**Pull request merged**